examples/mask_rcnn/train_mask_rcnn.py

```diff
@@ -0,0 +1,252 @@
+
+
+import functools
+import random
+import time
+import datetime
+
+from torchvision import transforms as T
+from models.mask_rcnn import MaskRCNN, Resize, Normalize
+from models.resnet import ResNet
+
+def build_transforms(is_train=True):
+  pixel_mean = [102.9801, 115.9465, 122.7717]
+  pixel_std = [1., 1., 1.]
+  to_bgr255 = True
+  if is_train:
+    min_size = (800,)
+    max_size = 1333
+    flip_prob = 0.5
+  else:
+    min_size = (800,)
+    max_size = 1333
+    flip_prob = 0
+
+  normalize_transform = Normalize(
+    mean=pixel_mean, std=pixel_std, to_bgr255=to_bgr255
+  )
+
+  return T.Compose(
+    [
+      Resize(min_size, max_size),
+      T.ToTensor(), ## TODO: RandomHorizontalFlip (flip_prob) helps here for training
+      Normalize(
+          mean=pixel_mean, std=pixel_std, to_bgr255=True
+      ),
+      normalize_transform
+    ]
+  )
+
+
+def mlperf_log_epoch_start(iteration, iters_per_epoch):
+  # First iteration:
+  #     Note we've started training & tag first epoch start
+  if iteration == 0:
+    print("epoch start epoch 0")
+    return
+  if iteration % iters_per_epoch == 0:
+    epoch = iteration // iters_per_epoch
+    print("epoch start epoch {}".format(epoch))
+
+def make_data_sampler(dataset, shuffle, distributed):
+  # TODO Implement for tiny, this is an import data sampler
+  # if distributed:
+  #     return samplers.DistributedSampler(dataset, shuffle=shuffle)
+  # if shuffle:
+  #     sampler = torch.utils.data.sampler.RandomSampler(dataset)
+  # else:
+  #     sampler = torch.utils.data.sampler.SequentialSampler(dataset)
+  return None
+
+def make_data_loader(cfg,
+                num_gpus=1,
+                images_per_batch=1,
+                start_iter=0, random_number_generator=None):
+  images_per_batch = cfg.SOLVER.IMS_PER_BATCH
+  assert (
+    images_per_batch % num_gpus == 0
+  ), "SOLVER.IMS_PER_BATCH ({}) must be divisible by the number "
+  "of GPUs ({}) used.".format(images_per_batch, num_gpus)
+  images_per_gpu = images_per_batch // num_gpus
+  shuffle = True
+  num_iters = 90000
+
+  # group images which have similar aspect ratio. In this case, we only
+  # group in two cases: those with width / height > 1, and the other way around,
+  # but the code supports more general grouping strategy
+  aspect_grouping = [1]
+
+  # todo reintroduce test 
+  dataset_list = ["/path/to/train"]
+
+  transforms = build_transforms(is_train=True)
+
+  ## TODO, this needs a tiny equivalent
+  # datasets, epoch_size = build_dataset(dataset_list, transforms, DatasetCatalog, is_train)
```

Comment:
This "dataset" api doesn't appear to have a tiny equivalent. Correct me if I'm wrong. 

For good training we have to have a util that quickly pre-reads random images from disk and serves them to the model. This is probably one of the best places for training time gains.